src/appleseed/foundation/platform/path.cpp

```diff
@@ -133,6 +137,44 @@ const char* get_executable_directory()
     return path;
 }
 
+const char* get_home_directory()
+{
+    static char path[FOUNDATION_MAX_PATH_LENGTH + 1];
+    static bool path_initialized = false;
+
+    if (!path_initialized)
+    {
+#if defined _WIN32
+
+        return 0;
+
++#elif defined __APPLE__
+
+        return 0;
+
+#elif defined __linux__ || defined __FreeBSD__
+
+        const char *home_dir = getenv("HOME");
+
+        if (home_dir == 0)
+            home_dir = getpwuid(getuid())->pw_dir;
+
+        strncpy(path, home_dir, FOUNDATION_MAX_PATH_LENGTH);
+        path[FOUNDATION_MAX_PATH_LENGTH] = 0;
```

Comment:
Since `path` is an array of `char`s, NULL-termination reads better as `= '\0';`.  This is quite common practice in C/C++ code that deals with strings.  It also helps compiler to do a better job on type checking.